lib/ast-to-react.js

```diff
@@ -141,6 +141,20 @@ export function childrenToReact(context, node) {
     } else if (child.type === 'raw' && !context.options.skipHtml) {
       // Default behavior is to show (encoded) HTML.
       children.push(child.value)
+    } else if (!context.options.skipHtml && context.options.components) {
+      const {type, ...props} = child
+      /** @type {string} */
+      const component =
+        context.options.components && own.call(context.options.components, type)
+          ? context.options.components[type]
+          : type
+      const hasCustomComponentType = Object.keys(
+        context.options.components
+      ).includes(type)
+      if (hasCustomComponentType) {
+        const key = [type, childIndex].join('-')
+        children.push(React.createElement(component, {...props, key}))
+      }
```

Comment:
components generally refer to an element type.
I think this would muddle/confuse that by being either: an element type or a node type.